File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/provider/TableEndpoint.java
package com.raizlabs.android.dbflow.annotation.provider;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description: Defines an endpoint that gets placed inside of a {@link com.raizlabs.android.dbflow.annotation.provider.ContentProvider}
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface TableEndpoint {

    /**
     * @return The name of the table this endpoint corresponds to.
     */
    String name();

    /**
     * @return When placed in a top-level class, this is required to connect it to a provider.
     */
    String contentProviderName() default "";
}

File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/provider/TableEndpoint.kt
package com.raizlabs.android.dbflow.annotation.provider

import kotlin.reflect.KClass

/**
 * Description: Defines an endpoint that gets placed inside of a [ContentProvider]
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.SOURCE)
annotation class TableEndpoint(
        /**
         * @return The name of the table this endpoint corresponds to.
         */
        val name: String,
        /**
         * @return When placed in a top-level class, this is required to connect it to a provider.
         */
        val contentProvider: KClass<*>)
